docs/docs/code/python/using-data-stores/README.md

```diff
@@ -84,7 +84,19 @@ def handler(pd: "pipedream"):
 
 ## Retrieving data
 
-You can retrieve data with the Data Store by key name:
+Data Stores are very performant at retrieving single records by keys. However you can also use key iteration to retrieve all records within a Data Store as well.
+
+::: tip
+
+Data Stores are intended to be a fast and convienent data storage option for quickly adding data storage capability to your workflows without adding another database dependency.
+
+However, if you need more advanced querying capabilities, consider using a full fledged database. Pipedream can integrate with MySQL, Postgres, DynamoDb, MongoDB and more.
```

Comment:
Can you expand on "advanced querying". Reading this as an outsider, I'm not sure what that means. Might be nice to differentiate briefly between key-value stores (where data stores are awesome) and relational databases / other DBs.

Reply:
Honestly a "When should I use Data Stores" section in the general Data store docs would be super useful so you can expand on this.

Reply:
Yes I was planning on adding FAQs to all resources, but just focusing on data store changes and python changes.

I would still like to leave this blurb here, because there are different terms across the languages, like describing nested dictionaries in a record would only be applicable in the Python docs.